src/main/java/History.java

```diff
@@ -1,49 +1,167 @@
      * commitID + " " + TIMESTAMP in the map build_history
+     *
      * @param buildResult, contains the result from the build
-     * @param commitID, unique ID for the commit/process/push should be searchable
+     * @param commitID,    unique ID for the commit/process/push should be searchable
      */
-    public void save(String buildResult, String commitID){
+    public void save(String buildResult, String commitID) {
 
     }
 
     /**
      * The method execute searchForCommitID(). If the file exists, the method returns the content of the commit.
      * Else it returns null.
+     *
      * @param commitID
      * @return content, which is the content of the commitID
      */
-    public String load(String commitID){
+    public String load(String commitID) {
         // if searchForCommits() returns null , return null
         return null;
     }
 
     /**
      * The function listHistory returns all the commitID in the directory build_history in execution order
+     *
      * @return, returns the history as a string, null if it does not exist
      */
-    public String listHistory(){
-        return null;
+    public String listHistory() throws Exception {
+        File dir = new File(directoryPath);
+        if (!dir.exists())
+            return "";
+
+        File[] files = dir.listFiles();
+        if (files == null)
+            throw new FileNotFoundException("Error in reading files");
+
+        ArrayList<CommitIDAndTimeStampHolder> holders = new ArrayList<CommitIDAndTimeStampHolder>();
+
+        for (File file : files) {
+            String[] info = file.getName().split(" ");
+            if (info.length != 2)
+                throw new Exception("Wrong amount of inputs");
+            CommitIDAndTimeStampHolder commitIDAndTimeStampHolder = new CommitIDAndTimeStampHolder(info[0], info[1]);
+            holders.add(commitIDAndTimeStampHolder);
+        }
+        Collections.sort(holders, commitIDAndTimeStampHolderComparator);
+
+        StringBuilder stringBuilder = new StringBuilder();
+        for (CommitIDAndTimeStampHolder holder : holders) {
+            stringBuilder.append(holder.getCommitID() + " " + holder.getTimeStamp());
+            stringBuilder.append("\n");
+        }
+
+        return stringBuilder.toString();
     }
 
     /**
      * The function checks if the file containing the name commiteID exists and return the file, otherwise null
+     *
      * @param commitID
      * @return the file with correct ID or null if it does not exist
      */
-    private File searchForCommitID(String commitID){
+    public File searchForCommitID(String commitID) {
+        File dir = new File(directoryPath);
+        if (!dir.exists())
+            return null;
+
+        File[] files = dir.listFiles();
+        if (files == null)
+            return null;
+
+        for (File file : files) {
+            String[] info = file.getName().split(" ");
+            if (info.length != 2)
+                return null;
```

Comment:
Shouldn't this line be `continue;` instead of `return null;` (because the loop should continue searching even if there was an error in one file)?

Reply:
Will fix that, seems better. I added return null to signal that the database is corrupted in some sort of way.